src/bin/lychee/main.rs

```diff
@@ -118,21 +118,22 @@ async fn run(cfg: &Config, inputs: Vec<Input>) -> Result<i32> {
         .accepted(accepted)
         .build()?;
 
-    let links = collector::collect_links(
-        &inputs,
+    let mut links = collector::collect_links(
+        inputs,
         cfg.base_url.clone(),
         cfg.skip_missing,
-        max_concurrency,
+        cfg.max_concurrency,
     )
     .await?;
 
     let pb = match cfg.no_progress {
         true => None,
         false => {
-            let bar = ProgressBar::new(links.len() as u64)
-                .with_style(ProgressStyle::default_bar().template(
+            let bar = ProgressBar::new_spinner().with_style(ProgressStyle::default_bar().template(
                 "{spinner:.red.bright} {pos}/{len:.dim} [{elapsed_precise}] {bar:25} {wide_msg}",
             ));
+            bar.set_length(0);
```

Comment:
Noticed that the progress bar is a bit "bumpy" in the beginning. (There's a better word for what I mean, but I can't seem to find it, heh.)
Keeping the initial length of `links.len()` is not possible, of course because at this point we don't have the length (since `links` is a channel now, not a `HashSet`), but I wonder if there is another way to avoid that "bumpy" behavior. Maybe we can get the initial length from the channel somehow to give a better estimate?

Reply:
Are you talking about the initial period where no links were extracted yet? Or that, when links come streaming in, the progress bar can jump around?
I don't see a way to get the amount of current messages already waiting in the stream, did you mean using the capacity of the stream as the initial value?

Reply:
Yeah, I meant the jumping when links start streaming in. Thought there was a `length` method for `mpsc`, but it doesn't exist. Probably mixed it up with another channel impl. Anyway, if we can't find a solution for the jumpiness, it's fine, too. 🤷‍♀️

Reply:
The downside of concurrency 😄. I think its fine when the bar represents the `n/m` process/gathered URLs in the front, else it may be even more confusing. Only other idea would be to not show any bar until all inputs have been processed, but this example (video in OP) shows that sometimes the checker is waiting for the extractor, so no process bar at all then 😢.

Should be fine as it is.